### DISCUSSION TITLE
supply a list via command line parameters

### DISCUSSION THREAD

**Discussion by seanrjohnson** [nextflow-io/nextflow#2821] [Q&A] [answered]
I would like to do something like this:
`nextflow main.nf --foo param1 param2 param3`

which would have the effect:
`params.foo = ["param1", "param2", "param3"]`

Is there a way to do that? I don't want to use wildcards, because they might not be files, or if they are files, they might not be named in a regular pattern.

Thanks

**Comment by jorgeaguileraseqera** [comment, marked as answer]
As a workaround, you can pass the `foo` argument as a String with commas and create a new params `fooList` splitting it.

`nextflow main.nf --foo param1,param2,param3`

main.nf :
```
params.foolist = params.foo?.split(',') as List
```

**Reply by seanrjohnson** [reply]
Thanks. That's a reasonable solution.
